x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsPhysicalOperationProviders.java

```diff
@@ -166,7 +169,25 @@ public final PhysicalOperation sourcePhysicalOperation(EsQueryExec esQueryExec,
         assert esQueryExec.estimatedRowSize() != null : "estimated row size not initialized";
         int rowEstimatedSize = esQueryExec.estimatedRowSize();
         int limit = esQueryExec.limit() != null ? (Integer) esQueryExec.limit().fold() : NO_LIMIT;
-        if (sorts != null && sorts.isEmpty() == false) {
+        if (EsqlCapabilities.Cap.METADATA_SCORE.isEnabled() && esQueryExec.scoring()) {
+            if (sorts != null) {
+                fieldSorts = new ArrayList<>(sorts.size());
+                for (FieldSort sort : sorts) {
+                    fieldSorts.add(sort.fieldSortBuilder());
+                }
+            } else {
+                fieldSorts = List.of();
+            }
+            luceneFactory = new ScoringLuceneTopNSourceOperator.Factory(
```

Comment:
Is there a way to decouple the sort and score? If users don't specify a sort explicitly, we don't need to pay the overhead of sort.

Reply:
Resolved.